test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -277,10 +278,10 @@ class test_ucp_sockaddr : public ucp_test {
         test_ucp_sockaddr *self = reinterpret_cast<test_ucp_sockaddr*>(arg);
         self->err_handler_count++;
         /* The current expected errors are only from the err_handle test
-         * and from transports where the worker address is too long  */
-        /* TODO: once large worker address is supported, and the error handling
-         * requirement is removed, only UCS_ERR_UNREACHABLE should be handled here */
-        if ((status != UCS_ERR_UNREACHABLE) && (status != UCS_ERR_BUFFER_TOO_SMALL)) {
+         * and from transports where the worker address is too long but ud/ud_x
```

Comment:
maybe skip the test earlier if ud/ud_x are not present?
what is "too long ud/ud_x errors"?

Reply:
well detecting the lack of ud/ud_x is part of the error handling flow and it will be skipped.
in the comment i meant error for  when the transports used for sending partial worker address are not present, i will clarify this doc.

Reply:
My point is the tests shouldn't skip real issues. pls keep only actual err codes, I still don't understand all of them.

Reply:
These are actual valid error codes.
There might be a case where the ud/ud_x transports are not present at all. It depends on the used transports for the test. In case it happens, it is a valid error but the test should not fail, but skipped.
It is also possible that the ud/ud_x trasnports are present, but their own addresses are too long and then it is also a valid error but the test should not fail on this - it depends on the setup.

Reply:
So, user can have setup where the functionality does not work, do user have a way to detect/handle it before everything is started?

Reply:
if the user doesnt have ud/ud_x, we (currently) cannot handle a large worker address. we exit gracefully with an error.
can't handle it before according to current design.

Reply:
My question was about "the ud/ud_x trasnports are present, but their own addresses are too long" + UCT_IFACE_FLAG_SUPPORT_PRESENT + rdms_cm capabilities.
If i understand correctly, these conditions can be checked locally and if successful, it should be enough for startup.

Reply:
IMHO it is better to check how ucx would handle these scenarios, with err handle, and skip during the test, not before. Skipping before doesn't check the error code.

Reply:
Ok, when does `UCS_ERR_BUFFER_TOO_SMALL` happen? Is it internal buffer for sending UD address? if yes, it should be `UCS_ERR_UNREACHABLE`.
On other hand, can iface has `UCT_IFACE_FLAG_SUPPORT_PRESENT` capability flag if its address too large for internal exchange? Seems like we need some cross TL dependency resolution mechanism on UCP level: rdma_cm->ud->rdma_cm
> IMHO it is better to check how ucx would handle these scenarios

For user, it's better to check all caps before start and fallback to something else as early as possible.

Reply:
i will update these errors to return UCS_ERR_UNREACHABLE.
we cannot detect this before since the importance of ud/ud_x lack is only relevant if the worker's address is too long, which depends on the sockaddr transport limitation, which is selected depending on the destination.